source/stylesheets/components/_header.scss

```diff
@@ -0,0 +1,113 @@
+// Known issues:
+//
+// When the GOV.UK logotype is focussed in Firefox, the outline is drawn based
+// on a different box to the background, which leads to 'gaps' on the top and
+// bottom between the outline and the background.
+//
+// This seems to only happen when the link is displayed 'inline-block', but
+// removing this affects the positioning of the underline and causes other
+// issues with the display of the focus style in IE8-11.
+
+@include exports("header") {
+
+  .app-c-header {
+    padding: $govuk-spacing-scale-2 $govuk-spacing-scale-3;
+    border-bottom: 10px solid $govuk-blue;
+    color: $govuk-white;
+    background: $govuk-black;
+  }
+
+  .app-c-header--logotype {
+    margin-right: $govuk-spacing-scale-1;
+  }
+
+  .app-c-header--logotype--crown {
+    margin-right: 1px;
+    fill: currentColor;
+    vertical-align: middle;
+  }
+
+  .app-c-header--logotype--crown--fallback-image {
+    width: 36px;
+    height: 32px;
+    border: 0;
+    vertical-align: middle;
+  }
+
+  .app-c-header--title {
+    @include govuk-core-24;
+    @include font-smoothing;
+  }
+
+  .app-c-header--link {
+    // Font size needs to be set on the link so that the box sizing is correct
+    // in Firefox 
+    @include _core-font-generator(
+      $font-size: 30px,
+      $font-size-640: 30px,
+      $font-size-print: 30px,
+      $line-height: 30px,
+      $line-height-640: 30px,
+      $font-weight: 700
+    );
+    @include font-smoothing;
+
+    display: inline-block;
+
+    &:link,
+    &:visited {
+      color: $govuk-white;
+      text-decoration: none;
+    }
+
+    &:hover,
+    &:active {
+      margin-bottom: -1px;
+      border-bottom: 1px solid $govuk-white;
+    }
+
+    &:focus {
+      color: $govuk-black;
```

Comment:
Shall we add `border-color: $govuk-black` on focus? Right now it looks like in the image below, but it would be good to have the border in the same color with the text all the time.
<img width="337" alt="screen shot 2017-11-07 at 09 40 01" src="https://user-images.githubusercontent.com/788096/32486697-516310c6-c39f-11e7-9f5c-d8d43dfe7ad2.png">